## Набор данных №1:

In [14]:
from tqdm import tqdm

headlines_f1 =['Идентификатор строки', 'Месяц признания безработным', 'Номер недели признания безработным', 'Пол', 'Возраст', 'Категория лиц, испытывающих трудности (отнесенность к одной или нескольким категориям)', 'Категория лиц, испытывающих трудности: «Повторно обратившийся»', 'Категория лиц, испытывающих трудности: «Ранее не работавшие»', 'Категория лиц, испытывающих трудности: «Предпенсионный возраст»', 'Категория лиц, испытывающих трудности: «Многодетные родители»', 'Категория лиц, испытывающих трудности: «Завершившие обучение по направлению службы занятости»', 'Категория лиц, испытывающих трудности: «Разведен (разведена)»', 'Категория лиц, испытывающих трудности: «Одинокие родители»', 'Категория лиц, испытывающих трудности: «Высвобождаемые работники»', 'Категория лиц, испытывающих трудности: «Малоимущие граждане, заключившие социальный контракт»', 'Категория (основание) незанятости', 'Месяц и год наступления категории незанятости', 'Месяц прекращения занятости', 'Уровень образования', 'Год выпуска из учебного заведения (последнего)', 'Профессия по последнему образованию', 'Регион', 'Федеральный округ', 'Трудовой стаж за последние 12 месяцев в количестве интервалов по 5 недель', 'Средняя заработная плата за последние 3 месяца по последнему месту работы в рублях', 'Профессия по последнему месту работы', 'Причина увольнения с последнего места работы' , 'Месяц увольнения с последнего места работы', 'Факт направления на профессиональное обучение', 'Специальность профессионального обучения', 'Продолжительность профессионального обучения', 'Количество направлений на работу', 'Количество отказов клиента от направления на работу', 'Количество отказов от подходящей работы', 'Количество отказов работодателя', 'Размер желаемой заработной платы', 'Характер работы трудоустройства', 'Режим работы трудоустройства', 'Месяц трудоустройства', 'Неделя трудоустройства', 'Профессия трудоустройства', 'Месяц закрытия (снятия с учета в качестве ищущего работу)', 'Причина снятия гражданина с учета в качества ищущего работу', 'Месяц завершения работы с гражданином', 'Причина завершения работы с гражданином']


# Расшифровка параметра age - возраст гражданина на момент подачи заявления. Возраст сгруппирован в следующие интервалы:
age_encode = {1 : '16-17 лет',
2 : 'от 18 до 22 лет включительно',
3 : 'от 23 до 27 лет включительно',
4 : 'от 28 до 32 лет включительно',
5 : 'от 33 до 37 лет включительно',
6 : 'от 38 до 42 лет включительно',
7 : 'от 43 до 47 лет включительно',
8 : 'от 48 до 52 лет включительно',
9 : '53-54 года',
10 : '55 лет',
11 : '56 лет',
12 : '57 лет'}

counter = 0
month_of_unemployment_statistics = {}

def int_convert(string): 
        try:
            if isinstance(float(string), float) and float(string) - int(float(string)) == 0.00:
                return int(float(string))
            elif isinstance(float(string), float) and float(string) - int(string) != 0:
                return float(string)
            elif isinstance(int(string), int):
                return int(string)
            elif isinstance(bool(string), bool):
                return bool(string)
        except:
            return string

f1 = open('datasets/unemployed_1_data.csv', encoding='utf-8')
for line in tqdm(f1):
    if counter == 0: 
        head = line.strip().split(';')
    else:
        dataline = line.strip().split(';')
        for i in range(len(dataline)):
            dataline[i] = int_convert(dataline[i])
            try: 
                month_of_unemployment_statistics[dataline[1]] += 1
            except:
                month_of_unemployment_statistics[dataline[1]] = 1
            
    counter+=1

print(month_of_unemployment_statistics)


1it [00:00, 996.51it/s]


KeyError: 'Самарская область'

In [12]:
print(headlines_f1.index('Регион'))

21


In [154]:
for i,n in enumerate(headlines_f1):
    print(i,n)

0 Идентификатор строки
1 Месяц признания безработным
2 Номер недели признания безработным
3 Пол
4 Возраст
5 Категория лиц, испытывающих трудности (отнесенность к одной или нескольким категориям)
6 Категория лиц, испытывающих трудности: «Повторно обратившийся»
7 Категория лиц, испытывающих трудности: «Ранее не работавшие»
8 Категория лиц, испытывающих трудности: «Предпенсионный возраст»
9 Категория лиц, испытывающих трудности: «Многодетные родители»
10 Категория лиц, испытывающих трудности: «Завершившие обучение по направлению службы занятости»
11 Категория лиц, испытывающих трудности: «Разведен (разведена)»
12 Категория лиц, испытывающих трудности: «Одинокие родители»
13 Категория лиц, испытывающих трудности: «Высвобождаемые работники»
14 Категория лиц, испытывающих трудности: «Малоимущие граждане, заключившие социальный контракт»
15 Категория (основание) незанятости
16 Месяц и год наступления категории незанятости
17 Месяц прекращения занятости
18 Уровень образования
19 Год выпуска из

In [5]:
#month_of_unemployment_statistics = {'2020-08': 24333570, '2020-05': 38490525, '2020-03': 6753285, '2020-06': 35310645, '2020-07': 29809800, '2020-04': 42716925, '2020-09': 17145090}

## Считаем количество мужчин и женщин, обратившихся в службу занятости по отдельным регионам:

In [48]:
gender_by_region = {}
counter = 0
f1 = open('datasets/unemployed_1_data.csv', encoding='utf-8')
for line in tqdm(f1):
    if counter == 0: 
        head = line.strip().split(';')
    else:
        dataline = line.strip().split(';')
        if dataline[21] in gender_by_region.keys():
            if dataline[3] in gender_by_region[dataline[21]].keys():
                 gender_by_region[dataline[21]][dataline[3]] +=1
            else:
                 gender_by_region[dataline[21]][dataline[3]] = 1
        else:
            gender_by_region[dataline[21]] = {}
            gender_by_region[dataline[21]][dataline[3]] = 1
    counter+=1  
    #if counter > 500:
    #    break
            
    

print(gender_by_region)

4323553it [00:34, 124332.65it/s]{'Самарская область': {'Женский': 47582, 'Мужской': 32559}, 'Ставропольский край': {'Мужской': 35034, 'Женский': 54436}, 'Иркутская область': {'Мужской': 31855, 'Женский': 45562}, 'г. Санкт-Петербург': {'Мужской': 49037, 'Женский': 79270}, 'Новосибирская область': {'Мужской': 38727, 'Женский': 54211}, 'Республика Северная Осетия - Алания': {'Женский': 15581, 'Мужской': 10797}, 'Республика Дагестан': {'Женский': 76833, 'Мужской': 64370}, 'г. Москва': {'Женский': 148700, 'Мужской': 109001}, 'Республика Татарстан (Татарстан)': {'Женский': 60005, 'Мужской': 39481}, 'Псковская область': {'Мужской': 6429, 'Женский': 8151}, 'Забайкальский край': {'Мужской': 19027, 'Женский': 24182}, 'Калининградская область': {'Женский': 26791, 'Мужской': 17503}, 'Ростовская область': {'Женский': 80833, 'Мужской': 52411}, 'Московская область': {'Женский': 109738, 'Мужской': 69724}, 'Орловская область': {'Женский': 9836, 'Мужской': 6901}, 'Республика Тыва': {'Женский': 13928, 'М

In [51]:
atribute_1 = 21
atribute_2 = 1
dict_of_atrubutes = {}

counter = 0
f1 = open('datasets/unemployed_1_data.csv', encoding='utf-8')
for line in tqdm(f1):
    if counter == 0: 
        head = line.strip().split(';')
    else:
        dataline = line.strip().split(';')
        (dataline[i]int_convert(dataline[i])
        if dataline[atribute_1] in dict_of_atrubutes.keys():
            if dataline[atribute_2] in dict_of_atrubutes[dataline[atribute_1]].keys():
                 dict_of_atrubutes[dataline[atribute_1]][dataline[atribute_2]] +=1
            else:
                 dict_of_atrubutes[dataline[atribute_1]][dataline[atribute_2]] = 1
        else:
            dict_of_atrubutes[dataline[atribute_1]] = {}
            dict_of_atrubutes[dataline[atribute_1]][dataline[atribute_2]] = 1
    counter+=1  
    #if counter > 500:
    #    break
            
    

print(dict_of_atrubutes)

4323553it [00:34, 124768.22it/s]{'Самарская область': {'2020-08': 9213, '2020-06': 15490, '2020-04': 13706, '2020-05': 21244, '2020-07': 11068, '2020-09': 7017, '2020-03': 2403}, 'Ставропольский край': {'2020-08': 11933, '2020-04': 14757, '2020-07': 15978, '2020-09': 8646, '2020-06': 16714, '2020-05': 19687, '2020-03': 1755}, 'Иркутская область': {'2020-05': 15460, '2020-09': 8726, '2020-06': 12333, '2020-03': 3474, '2020-07': 12993, '2020-04': 13142, '2020-08': 11289}, 'г. Санкт-Петербург': {'2020-08': 13096, '2020-05': 25744, '2020-04': 43425, '2020-06': 18826, '2020-09': 9066, '2020-07': 14265, '2020-03': 3885}, 'Новосибирская область': {'2020-03': 3741, '2020-06': 17528, '2020-08': 12921, '2020-05': 17698, '2020-07': 14907, '2020-09': 10058, '2020-04': 16085}, 'Республика Северная Осетия - Алания': {'2020-06': 5680, '2020-07': 6405, '2020-04': 2637, '2020-05': 3746, '2020-08': 3829, '2020-09': 3122, '2020-03': 959}, 'Республика Дагестан': {'2020-07': 37923, '2020-08': 37588, '2020-

In [161]:
f1.close()

Распределение месяцев признания безработным по регионам: {'Самарская область': {'2020-08': 9213, '2020-06': 15490, '2020-04': 13706, '2020-05': 21244, '2020-07': 11068, '2020-09': 7017, '2020-03': 2403}, 'Ставропольский край': {'2020-08': 11933, '2020-04': 14757, '2020-07': 15978, '2020-09': 8646, '2020-06': 16714, '2020-05': 19687, '2020-03': 1755}, 'Иркутская область': {'2020-05': 15460, '2020-09': 8726, '2020-06': 12333, '2020-03': 3474, '2020-07': 12993, '2020-04': 13142, '2020-08': 11289}, 'г. Санкт-Петербург': {'2020-08': 13096, '2020-05': 25744, '2020-04': 43425, '2020-06': 18826, '2020-09': 9066, '2020-07': 14265, '2020-03': 3885}, 'Новосибирская область': {'2020-03': 3741, '2020-06': 17528, '2020-08': 12921, '2020-05': 17698, '2020-07': 14907, '2020-09': 10058, '2020-04': 16085}, 'Республика Северная Осетия - Алания': {'2020-06': 5680, '2020-07': 6405, '2020-04': 2637, '2020-05': 3746, '2020-08': 3829, '2020-09': 3122, '2020-03': 959}, 'Республика Дагестан': {'2020-07': 37923, '2020-08': 37588, '2020-09': 21436, '2020-04': 9913, '2020-05': 7323, '2020-06': 22454, '2020-03': 4566}, 'г. Москва': {'2020-08': 23151, '2020-07': 27256, '2020-05': 70139, '2020-06': 39610, '2020-04': 72845, '2020-03': 7051, '2020-09': 17649}, 'Республика Татарстан (Татарстан)': {'2020-03': 3198, '2020-07': 11861, '2020-04': 34974, '2020-06': 15399, '2020-05': 18118, '2020-08': 9828, '2020-09': 6108}, 'Псковская область': {'2020-04': 2866, '2020-05': 3924, '2020-08': 1355, '2020-06': 2837, '2020-09': 1159, '2020-07': 1916, '2020-03': 523}, 'Забайкальский край': {'2020-03': 1415, '2020-07': 7620, '2020-05': 8731, '2020-06': 9458, '2020-09': 4120, '2020-08': 5124, '2020-04': 6741}, 'Калининградская область': {'2020-09': 3873, '2020-04': 11311, '2020-06': 7778, '2020-07': 6987, '2020-08': 5202, '2020-05': 8042, '2020-03': 1101}, 'Ростовская область': {'2020-04': 29333, '2020-09': 15256, '2020-06': 21356, '2020-08': 19036, '2020-05': 24083, '2020-07': 20292, '2020-03': 3888}, 'Московская область': {'2020-05': 41169, '2020-08': 16593, '2020-04': 54747, '2020-06': 26755, '2020-07': 20313, '2020-09': 15149, '2020-03': 4736}, 'Орловская область': {'2020-05': 3345, '2020-08': 1490, '2020-04': 5382, '2020-07': 2031, '2020-06': 2644, '2020-09': 1101, '2020-03': 744}, 'Республика Тыва': {'2020-06': 6679, '2020-05': 5005, '2020-08': 2515, '2020-07': 5576, '2020-04': 5898, '2020-09': 1445, '2020-03': 1359}, 'Кемеровская область': {'2020-09': 12532, '2020-04': 14876, '2020-05': 17996, '2020-08': 16202, '2020-07': 15806, '2020-06': 15477, '2020-03': 4670}, 'Пензенская область': {'2020-05': 7560, '2020-07': 5998, '2020-08': 4840, '2020-06': 6779, '2020-04': 9844, '2020-09': 3892, '2020-03': 946}, 'Волгоградская область': {'2020-05': 13807, '2020-08': 5985, '2020-07': 8285, '2020-06': 11366, '2020-04': 13145, '2020-03': 1774, '2020-09': 3822}, 'Пермский край': {'2020-07': 12024, '2020-06': 12819, '2020-04': 24210, '2020-05': 13423, '2020-08': 8702, '2020-09': 5957, '2020-03': 3354}, 'Ярославская область': {'2020-06': 6616, '2020-05': 7897, '2020-04': 11389, '2020-07': 5124, '2020-03': 1729, '2020-09': 2898, '2020-08': 4044}, 'Республика Башкортостан': {'2020-03': 4586, '2020-08': 21544, '2020-06': 26878, '2020-05': 23160, '2020-07': 23587, '2020-09': 13824, '2020-04': 36013}, 'Тюменская область': {'2020-07': 4566, '2020-05': 9952, '2020-04': 16196, '2020-08': 2939, '2020-06': 8980, '2020-09': 2555, '2020-03': 1292}, 'Томская область': {'2020-06': 5662, '2020-05': 7595, '2020-07': 5618, '2020-04': 5051, '2020-08': 5019, '2020-09': 3652, '2020-03': 1122}, 'Республика Саха (Якутия)': {'2020-08': 5343, '2020-04': 9202, '2020-06': 5624, '2020-05': 5772, '2020-03': 1554, '2020-07': 5346, '2020-09': 1258}, 'Смоленская область': {'2020-09': 2216, '2020-06': 4962, '2020-05': 4900, '2020-08': 3035, '2020-07': 4325, '2020-04': 4827, '2020-03': 1118}, 'Омская область': {'2020-08': 9674, '2020-07': 11775, '2020-06': 11803, '2020-04': 10843, '2020-05': 13551, '2020-09': 5564, '2020-03': 2357}, 'Свердловская область': {'2020-05': 28052, '2020-09': 12201, '2020-06': 22596, '2020-04': 33100, '2020-07': 19996, '2020-03': 5611, '2020-08': 17043}, 'Ханты-Мансийский автономный округ - Югра': {'2020-04': 7605, '2020-08': 3251, '2020-05': 8313, '2020-07': 5951, '2020-09': 2837, '2020-06': 7690, '2020-03': 987}, 'Краснодарский край': {'2020-06': 34103, '2020-07': 22085, '2020-05': 39882, '2020-04': 39737, '2020-08': 16625, '2020-09': 9452, '2020-03': 3608}, 'Новгородская область': {'2020-09': 1767, '2020-04': 3380, '2020-08': 1941, '2020-05': 3066, '2020-06': 2640, '2020-07': 2325, '2020-03': 601}, 'Курганская область': {'2020-07': 4976, '2020-09': 3754, '2020-05': 4764, '2020-06': 4952, '2020-04': 5657, '2020-08': 4331, '2020-03': 864}, 'Тверская область': {'2020-07': 3895, '2020-08': 3257, '2020-06': 4689, '2020-09': 2423, '2020-04': 5699, '2020-05': 5641, '2020-03': 969}, 'Ульяновская область': {'2020-08': 5448, '2020-09': 3810, '2020-05': 7054, '2020-07': 6372, '2020-06': 6884, '2020-04': 9660, '2020-03': 624}, 'Тамбовская область': {'2020-06': 2448, '2020-05': 2781, '2020-07': 1880, '2020-08': 1281, '2020-04': 2815, '2020-03': 594, '2020-09': 1004}, 'Кировская область': {'2020-06': 6442, '2020-07': 5282, '2020-05': 9251, '2020-09': 3733, '2020-04': 6145, '2020-08': 4182, '2020-03': 1378}, 'Саратовская область': {'2020-08': 10381, '2020-05': 13653, '2020-04': 13295, '2020-07': 12437, '2020-06': 13285, '2020-09': 7525, '2020-03': 2206}, 'Чувашская Республика - Чувашия': {'2020-05': 6836, '2020-04': 14466, '2020-03': 1497, '2020-06': 5861, '2020-08': 3808, '2020-07': 4652, '2020-09': 2247}, 'Республика Крым': {'2020-05': 12661, '2020-06': 9420, '2020-07': 7690, '2020-04': 13313, '2020-09': 6774, '2020-08': 8310, '2020-03': 965}, 'Астраханская область': {'2020-07': 7835, '2020-08': 6626, '2020-06': 8524, '2020-09': 4504, '2020-04': 9199, '2020-05': 7167, '2020-03': 1001}, 'Белгородская область': {'2020-08': 2264, '2020-05': 5566, '2020-06': 4266, '2020-04': 5884, '2020-07': 3540, '2020-09': 416, '2020-03': 839}, 'Республика Хакасия': {'2020-06': 2402, '2020-07': 2525, '2020-08': 1947, '2020-03': 877, '2020-04': 4073, '2020-09': 1788, '2020-05': 2246}, 'Красноярский край': {'2020-06': 17094, '2020-04': 21220, '2020-07': 14468, '2020-09': 7762, '2020-05': 16260, '2020-08': 10801, '2020-03': 2967}, 'Республика Ингушетия': {'2020-03': 2251, '2020-08': 9788, '2020-07': 19955, '2020-06': 13233, '2020-05': 2502, '2020-09': 4139, '2020-04': 1735}, 'Приморский край': {'2020-06': 9103, '2020-08': 5003, '2020-05': 14388, '2020-09': 3882, '2020-03': 2300, '2020-07': 5674, '2020-04': 3988}, 'Челябинская область': {'2020-06': 19024, '2020-07': 17119, '2020-09': 9986, '2020-05': 23652, '2020-04': 20012, '2020-03': 4522, '2020-08': 15128}, 'Республика Карелия': {'2020-04': 5978, '2020-06': 3570, '2020-05': 3727, '2020-08': 2015, '2020-07': 2896, '2020-03': 1233, '2020-09': 1676}, 'Кабардино-Балкарская Республика': {'2020-05': 11888, '2020-04': 12377, '2020-06': 16627, '2020-07': 15507, '2020-09': 4611, '2020-08': 8131, '2020-03': 1137}, 'Республика Бурятия': {'2020-04': 11175, '2020-09': 2104, '2020-06': 5340, '2020-03': 1441, '2020-07': 2951, '2020-08': 2553, '2020-05': 8347}, 'Удмуртская Республика': {'2020-04': 7013, '2020-06': 6950, '2020-07': 4978, '2020-09': 2563, '2020-08': 3734, '2020-05': 9530, '2020-03': 1681}, 'Ленинградская область': {'2020-05': 8914, '2020-07': 5324, '2020-09': 4356, '2020-04': 10210, '2020-06': 6567, '2020-08': 4567, '2020-03': 959}, 'Республика Калмыкия': {'2020-04': 2018, '2020-07': 2087, '2020-05': 1917, '2020-09': 1321, '2020-06': 2211, '2020-08': 1515, '2020-03': 400}, 'Чеченская Республика': {'2020-06': 56346, '2020-09': 11898, '2020-04': 16895, '2020-07': 23156, '2020-05': 23067, '2020-03': 9329, '2020-08': 22428}, 'Оренбургская область': {'2020-06': 13985, '2020-05': 14886, '2020-03': 2629, '2020-07': 14566, '2020-08': 12754, '2020-09': 8864, '2020-04': 10261}, 'Липецкая область': {'2020-04': 3328, '2020-06': 2134, '2020-05': 3810, '2020-09': 985, '2020-07': 2053, '2020-08': 1075, '2020-03': 573}, 'Карачаево-Черкесская Республика': {'2020-06': 9403, '2020-07': 7447, '2020-05': 3027, '2020-08': 3439, '2020-04': 1820, '2020-03': 521, '2020-09': 1793}, 'Сахалинская область': {'2020-04': 648, '2020-03': 425, '2020-07': 546, '2020-09': 436, '2020-05': 840, '2020-08': 498, '2020-06': 869}, 'Алтайский край': {'2020-07': 8089, '2020-09': 4338, '2020-05': 8771, '2020-06': 7549, '2020-08': 6781, '2020-04': 6718, '2020-03': 2542}, 'Республика Марий Эл': {'2020-04': 3272, '2020-09': 1265, '2020-06': 2660, '2020-08': 1493, '2020-07': 1856, '2020-05': 2611, '2020-03': 684}, 'Хабаровский край': {'2020-05': 6340, '2020-07': 4845, '2020-04': 4173, '2020-06': 5203, '2020-08': 4413, '2020-03': 1034, '2020-09': 3481}, 'Рязанская область': {'2020-08': 2447, '2020-05': 4615, '2020-04': 4451, '2020-07': 2788, '2020-06': 3654, '2020-03': 662, '2020-09': 1816}, 'Республика Коми': {'2020-09': 2361, '2020-08': 3069, '2020-06': 4416, '2020-04': 5956, '2020-07': 3969, '2020-05': 5210, '2020-03': 1701}, 'Камчатский край': {'2020-09': 736, '2020-04': 714, '2020-03': 391, '2020-08': 491, '2020-05': 810, '2020-07': 608, '2020-06': 661}, 'Брянская область': {'2020-04': 8857, '2020-06': 5552, '2020-05': 7905, '2020-07': 2396, '2020-09': 1725, '2020-08': 2154, '2020-03': 854}, 'Нижегородская область': {'2020-07': 10570, '2020-04': 17965, '2020-05': 16284, '2020-06': 13346, '2020-08': 8154, '2020-09': 6867, '2020-03': 1922}, 'Тульская область': {'2020-06': 4773, '2020-04': 7769, '2020-05': 5882, '2020-08': 3278, '2020-09': 2590, '2020-07': 4076, '2020-03': 916}, 'Курская область': {'2020-06': 4173, '2020-07': 3807, '2020-05': 4745, '2020-04': 6630, '2020-08': 2391, '2020-09': 1621, '2020-03': 767}, 'Амурская область': {'2020-05': 4614, '2020-08': 3179, '2020-09': 2137, '2020-03': 1491, '2020-04': 4136, '2020-06': 3990, '2020-07': 3845}, 'Ивановская область': {'2020-08': 3241, '2020-04': 9281, '2020-06': 4708, '2020-07': 3965, '2020-05': 6229, '2020-09': 2135, '2020-03': 678}, 'Мурманская область': {'2020-05': 3658, '2020-08': 1952, '2020-03': 1304, '2020-07': 2173, '2020-09': 2211, '2020-04': 3329, '2020-06': 2200}, 'Воронежская область': {'2020-09': 4730, '2020-06': 9735, '2020-05': 11606, '2020-04': 12232, '2020-07': 8246, '2020-08': 6796, '2020-03': 2235}, 'Республика Адыгея (Адыгея)': {'2020-05': 3251, '2020-07': 2357, '2020-04': 3315, '2020-09': 1451, '2020-06': 2851, '2020-08': 2188, '2020-03': 296}, 'Владимирская область': {'2020-04': 8601, '2020-06': 6170, '2020-05': 8782, '2020-07': 5121, '2020-09': 3145, '2020-08': 4090, '2020-03': 1551}, 'Еврейская автономная область': {'2020-05': 1027, '2020-06': 797, '2020-07': 826, '2020-08': 690, '2020-04': 899, '2020-09': 544, '2020-03': 242}, 'Архангельская область': {'2020-03': 1756, '2020-09': 3492, '2020-06': 5761, '2020-05': 5952, '2020-04': 7061, '2020-07': 5001, '2020-08': 3880}, 'Республика Мордовия': {'2020-08': 2221, '2020-04': 4222, '2020-06': 3002, '2020-07': 2575, '2020-05': 2965, '2020-09': 1534, '2020-03': 611}, 'Вологодская область': {'2020-07': 4991, '2020-04': 15218, '2020-03': 1338, '2020-09': 3346, '2020-08': 4083, '2020-05': 6941, '2020-06': 5051}, 'Калужская область': {'2020-06': 3056, '2020-04': 4943, '2020-07': 2395, '2020-05': 3919, '2020-09': 1457, '2020-08': 1784, '2020-03': 646}, 'Республика Алтай': {'2020-08': 2015, '2020-09': 1498, '2020-05': 2520, '2020-06': 2322, '2020-04': 2531, '2020-07': 2852, '2020-03': 537}, 'Ямало-Ненецкий автономный округ': {'2020-08': 1000, '2020-06': 1265, '2020-04': 2217, '2020-05': 1532, '2020-07': 1278, '2020-09': 965, '2020-03': 514}, 'Костромская область': {'2020-06': 3215, '2020-04': 6431, '2020-08': 2462, '2020-05': 3999, '2020-07': 2715, '2020-09': 1992, '2020-03': 596}, 'Ненецкий автономный округ': {'2020-04': 201, '2020-05': 220, '2020-07': 155, '2020-09': 208, '2020-08': 165, '2020-06': 197, '2020-03': 104}, 'г. Севастополь': {'2020-04': 1628, '2020-07': 419, '2020-05': 1760, '2020-06': 695, '2020-08': 334, '2020-09': 279, '2020-03': 136}, 'Чукотский автономный округ': {'2020-05': 103, '2020-09': 79, '2020-07': 130, '2020-03': 86, '2020-08': 88, '2020-04': 97, '2020-06': 154}, 'Магаданская область': {'2020-05': 335, '2020-08': 293, '2020-04': 354, '2020-06': 360, '2020-07': 379, '2020-09': 287, '2020-03': 190}}

In [156]:
def atribute_distribution(atribute_1 = 21, atribute_2 = 1):
    dict_of_atrubutes = {}
    counter = 0
    with open('datasets/unemployed_1_data.csv', encoding='utf-8') as f1:
        for line in tqdm(f1):
            if counter == 0: 
                head = line.strip().split(';')
            else:
                dataline = line.strip().split(';')
                dataline[atribute_1] = int_convert(dataline[atribute_1])
                dataline[atribute_2] = int_convert(dataline[atribute_2])

                if dataline[atribute_1] in dict_of_atrubutes.keys():
                    if dataline[atribute_2] in dict_of_atrubutes[dataline[atribute_1]].keys():
                        dict_of_atrubutes[dataline[atribute_1]][dataline[atribute_2]] +=1
                    else:
                        dict_of_atrubutes[dataline[atribute_1]][dataline[atribute_2]] = 1
                else:
                    dict_of_atrubutes[dataline[atribute_1]] = {}
                    dict_of_atrubutes[dataline[atribute_1]][dataline[atribute_2]] = 1
                #if counter == 50: 
                #    break
            counter+=1
    return dict_of_atrubutes 

print(atribute_distribution(21, 3))     

4323553it [00:48, 89229.41it/s]{'Самарская область': {'Женский': 47582, 'Мужской': 32559}, 'Ставропольский край': {'Мужской': 35034, 'Женский': 54436}, 'Иркутская область': {'Мужской': 31855, 'Женский': 45562}, 'г. Санкт-Петербург': {'Мужской': 49037, 'Женский': 79270}, 'Новосибирская область': {'Мужской': 38727, 'Женский': 54211}, 'Республика Северная Осетия - Алания': {'Женский': 15581, 'Мужской': 10797}, 'Республика Дагестан': {'Женский': 76833, 'Мужской': 64370}, 'г. Москва': {'Женский': 148700, 'Мужской': 109001}, 'Республика Татарстан (Татарстан)': {'Женский': 60005, 'Мужской': 39481}, 'Псковская область': {'Мужской': 6429, 'Женский': 8151}, 'Забайкальский край': {'Мужской': 19027, 'Женский': 24182}, 'Калининградская область': {'Женский': 26791, 'Мужской': 17503}, 'Ростовская область': {'Женский': 80833, 'Мужской': 52411}, 'Московская область': {'Женский': 109738, 'Мужской': 69724}, 'Орловская область': {'Женский': 9836, 'Мужской': 6901}, 'Республика Тыва': {'Женский': 13928, 'Му

## Функция сортировки вложенных словарей по значениям от большего к меньшему:

In [160]:
def sort_inner_dict(dict_1):
    new_dict1 = {}
    for key, dict_2 in dict_1.items():  
        sorted_dict2_values = sorted(dict_2.values(), reverse = True)    
        new_dict2 = {}
        for i in range(len(sorted_dict2_values)):
            
            for k, v in dict_2.items():
                if v == sorted_dict2_values[i]:
                    new_dict2[k] = sorted_dict2_values[i]
        new_dict1[key] = new_dict2 
    return new_dict1

#print(sort_inner_dict({'a':{'a1': 2, 'a2': 2, 'a3': 3}, 'b':{'b1': 40, 'b2': 4}}))
#print(atribute_distribution(21, 3))
print(sort_inner_dict(atribute_distribution(21, 3)))

4323553it [00:47, 91573.06it/s]{'Самарская область': {'Женский': 47582, 'Мужской': 32559}, 'Ставропольский край': {'Женский': 54436, 'Мужской': 35034}, 'Иркутская область': {'Женский': 45562, 'Мужской': 31855}, 'г. Санкт-Петербург': {'Женский': 79270, 'Мужской': 49037}, 'Новосибирская область': {'Женский': 54211, 'Мужской': 38727}, 'Республика Северная Осетия - Алания': {'Женский': 15581, 'Мужской': 10797}, 'Республика Дагестан': {'Женский': 76833, 'Мужской': 64370}, 'г. Москва': {'Женский': 148700, 'Мужской': 109001}, 'Республика Татарстан (Татарстан)': {'Женский': 60005, 'Мужской': 39481}, 'Псковская область': {'Женский': 8151, 'Мужской': 6429}, 'Забайкальский край': {'Женский': 24182, 'Мужской': 19027}, 'Калининградская область': {'Женский': 26791, 'Мужской': 17503}, 'Ростовская область': {'Женский': 80833, 'Мужской': 52411}, 'Московская область': {'Женский': 109738, 'Мужской': 69724}, 'Орловская область': {'Женский': 9836, 'Мужской': 6901}, 'Республика Тыва': {'Мужской': 14549, 'Же

### Доход до пандемии

In [265]:
#revenue = sort_inner_dict(atribute_distribution(21, 24))


def mean_revenue(atribute_to_compute_1 = 21, atribute_to_compute_2 = 24, what_to_compute= 'mean'):
    dict_of_values = {}
    amount_for_atributes = {} 
    count_for_compute = {} 
    counter = 0
    print(counter)
    with open('datasets/unemployed_1_data.csv', encoding='utf-8') as f1:
        for line in tqdm(f1):
            if counter == 0: 
                pass
            else:
                dataline = line.strip().split(';')
                dataline[atribute_to_compute_1] = int_convert(dataline[atribute_to_compute_1])
                dataline[atribute_to_compute_2] = int_convert(dataline[atribute_to_compute_2])
                
                if dataline[atribute_to_compute_2] == 'None' or dataline[atribute_to_compute_2] == 0:
                    continue
                else:
                    if dataline[atribute_to_compute_1] in amount_for_atributes.keys():
                        amount_for_atributes[dataline[atribute_to_compute_1]] += dataline[atribute_to_compute_2]
                        count_for_compute[dataline[atribute_to_compute_1]] += 1
                    else:
                        amount_for_atributes[dataline[atribute_to_compute_1]] = dataline[atribute_to_compute_2]
                        count_for_compute[dataline[atribute_to_compute_1]] = 1
            counter+=1
        
            #if counter > 500:
            #    break

    for key, val in amount_for_atributes.items():
        amount_for_atributes[key] = round(val/count_for_compute[key])

    sorted_amount = {}    
    sorted_values = sorted( amount_for_atributes.values(), reverse = True)
    
    for i in range(len(amount_for_atributes)):
        for k,v in  amount_for_atributes.items():
            if v==sorted_values[i]:
                sorted_amount[k] = sorted_values[i]  

    return sorted_amount
   
#revenue = mean_revenue()
#print(revenue)

with open('revenue.csv','w', encoding='utf-8') as revenue_file:
    revenue_file.write(headlines_f1[24] + '\n')
    for k,v in revenue.items():
        revenue_file.write('{} : {} \n'.format(k, v))

In [205]:
def mean_revenue(atribute_to_compute_1 = 21, atribute_to_compute_2 = 24, what_to_compute= 'mean'):
    dict_of_values = {}
    amount_for_atributes = {} 
    count_for_compute = {} 
    counter = 0
    print(counter)
    with open('datasets/unemployed_1_data.csv', encoding='utf-8') as f1:
        for line in tqdm(f1):
            if counter == 0: 
                pass
            elif counter > 5:
                break
            counter+=1
    return count_for_compute
mean_revenue()

6it [00:00, 6006.16it/s]0



{}

## Набор данных №2:

In [38]:
headlines_f2 =['Идентификатор строки','Источник подачи заявления', 'Код региона заявителя', 'Статус на момент формирования выгрузки', 'Причина снятия с учета', 'Код региона службы занятости населения', 'Признак отсутствия трудовой деятельности','Признак повторного заявления', 'Причина увольнения с последнего места работы', 'Наличие детей', 'Предпенсионер', 'Идентификатор пользователя', 'Работал в апреле 2019 года', 'Работал в мае 2019 года', 'Работал в июне 2019 года', 'Работал в июле 2019 года', 'Работал в августе 2019 года', 'Работал в сентябре 2019 года', 'Работал в октябре 2019 года', 'Работал в ноябре 2019 года', 'Работал в декабре 2019 года', 'Работал в январе 2020 года', 'Работал в феврале 2020 года', 'Работал в марте 2020 года', 'Работал в апреле 2020 года', 'Работал в мае 2020 года', 'Работал в июне 2020 года', 'Работал в июле 2020 года', 'Работал в августе 2020 года', 'Работал в сентябре 2020 года', 'Работал в октябре 2020 года', 'Средняя заработная плата за второй квартал 2019 года', 'Средняя заработная плата за второй квартал 2019 года', 'Средняя заработная плата за третий квартал 2019 года', 'Средняя заработная плата за четвертый квартал 2019 года', 'Средняя заработная плата за первый квартал 2020 года', 'Средняя заработная плата за второй квартал 2020 года', 'Средняя заработная плата за третий квартал 2020 года','Есть информация из резюме', 'Регион заявителя', 'Желаемая должность', 'Сфера деятельности', 'Уровень образования', 'Наличие водительского удостоверения', 'Опыт работы, лет', 'Ожидаемая заработная плата, рублей', 'Год рождения', 'Пол', 'Готовность к переобучению', 'Сумма выплат по пособию по безработице, рублей', 'Число дней между первой и последней выплатами пособия по безработице', 'Сумма выплат на детей', 'Месяц создания заявления', 'Неделя создания заявления', 'Месяц, в который принято решение о признании безработным', 'Неделя, в которую принято решение о признании безработным', 'Месяц трудоустройства', 'Неделя трудоустройства', 'Месяц отзыва заявления', 'Неделя отзыва заявления', 'Месяц снятия с учета', 'Неделя снятия с учета', 'Месяц, в который было принято решение об отказе в назначении пособия', 'Неделя, в которую было принято решения об отказе в назначении пособия']

#расшифровка значений status на наборе данных №2
status_encode = {'APPROVED' : 'Назначено пособие',
'DELETED' : 'Заявление отозвано',
'EMPLOYED' : 'Трудоустроен',
'FILLING_APPLICATION' : 'Дозаполнение/редактирование сведений (по просьбе ЦЗН)',
'IN_PROGRESS' : 'Заявление зарегистрировано в центре занятости населения',
'JOB_OFFERED' : 'Предложены вакансии',
'NEW' : 'Новое заявление',
'NOT_APPROVED' : 'Отказано в пособии',
'REMOVED' : 'Снят с учета',
'REQUEST' : 'Запрос сведений',
'REREGISTRATION' : 'Перерегистрация'}

counter = 0
data_sample2_dict ={}

with open('unemployed_2_sample.csv', encoding='utf-8') as f2:
    for line in f2:
        if counter == 0: 
            head = line.strip().split(';')
        elif counter <2:
            dataline = line.strip().split(';')
        else:
            break
        counter+=1

for name, value in zip(headlines_f2, dataline):
    data_sample2_dict[name] = value
    print("{} : {}".format(name, value))
 

Идентификатор строки : 1588b058-ccea-409b-9190-19f54f425e43
Источник подачи заявления : EPGU
Код региона заявителя : 50
Статус на момент формирования выгрузки : REREGISTRATION
Причина снятия с учета : null
Код региона службы занятости населения : 50
Признак отсутствия трудовой деятельности : false
Признак повторного заявления : false
Причина увольнения с последнего места работы : По собственному желанию (инициативе работника)
Наличие детей : true
Предпенсионер : false
Идентификатор пользователя : a355587a-c614-e1a6-c293-e53269bf4f92
Работал в апреле 2019 года : null
Работал в мае 2019 года : null
Работал в июне 2019 года : null
Работал в июле 2019 года : false
Работал в августе 2019 года : false
Работал в сентябре 2019 года : false
Работал в октябре 2019 года : false
Работал в ноябре 2019 года : false
Работал в декабре 2019 года : false
Работал в январе 2020 года : false
Работал в феврале 2020 года : false
Работал в марте 2020 года : false
Работал в апреле 2020 года : false
Работал в м

#### Загрузим таблицу соответствия названий регионов значению ОКТМО:

In [43]:
region_code = {}
with open('regions.csv') as oktmo_regions:
    for line in oktmo_regions:
        line_list = line.strip().split(';')
        if 'OKTMO_ID' in line:
            continue
        else:
            region_code[int(line_list[0])] = [line_list[1], line_list[2]] 
print (region_code)

{1: ['Алтайский край', 'Сибирский федеральный округ'], 3: ['Краснодарский край', 'Южный федеральный округ'], 4: ['Красноярский край', 'Сибирский федеральный округ'], 5: ['Приморский край', 'Дальневосточный федеральный округ'], 7: ['Ставропольский край', 'Северо–Кавказский федеральный округ'], 8: ['Хабаровский край', 'Дальневосточный федеральный округ'], 9: ['Белгородская область', 'Центральный федеральный округ'], 10: ['Амурская область', 'Дальневосточный федеральный округ'], 11: ['Ненецкий автономный округ', 'Северо-Западный федеральный округ'], 12: ['Астраханская область', 'Южный федеральный округ'], 14: ['Белгородская область', 'Центральный федеральный округ'], 15: ['Брянская область', 'Центральный федеральный округ'], 17: ['Владимирская область', 'Центральный федеральный округ'], 18: ['Волгоградская область', 'Южный федеральный округ'], 19: ['Вологодская область', 'Северо-Западный федеральный округ'], 20: ['Воронежская область', 'Центральный федеральный округ'], 22: ['Нижегородская